### PR TITLE
Run acceptance tests CircleCI

### PR DESCRIPTION
**Pull request by feedmypixel**
This work covers:

- [x] Running acceptance tests on `CircleCi`
- [x] Migrating to `CircleCi 2.0` workflows so we now have the following workflows:
  - `unit_tests`
  - `lint_code`
  - `acceptance_tests`
- [x] Providing html reports from `Nightwatch.js`
- [x] Move ~[feedmypixel/java-node-chrome](https://hub.docker.com/r/feedmypixel/java-node-chrome/)~ to `DIT` [docker-nightwatch-base](https://hub.docker.com/r/ukti/docker-nightwatch-base/) dockerhub
- [x] Move ~[feedmypixel/java-node-chrome-docker](https://github.com/feedmypixel/java-node-chrome-docker)~ to `DIT` [docker-nightwatch-base](https://github.com/uktrade/docker-nightwatch-base) GitHub
- [x] Comment out `acceptance_tests` job in CircleCI until backend test capacity is addressed.

## TODO
turn on required status checks for new jobs in GitHub

## Of Note
- As discussed I will comment out the `acceptance_tests` job until we can address the backend capacity issue.

- Currently Acceptance tests are not green so this job will not be a required status check until we have got on top of the tests and they are in a much more stable state

- GitHub repo [docker-nightwatch-base](https://github.com/uktrade/docker-nightwatch-base) created

- Docker Hub repo [docker-nightwatch-base](https://hub.docker.com/r/ukti/docker-nightwatch-base/) created

### Without Acceptance tests
screen shot 2017-09-18 at 13 08 31

### With Acceptance tests
screen shot 2017-09-18 at 08 28 33

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -267,6 +274,42 @@ layouts/
 +-----+ login.njk
 ```
 
+## Testing
+
+### Acceptance Testing
+data hub uses [Nightwatch.js](http://nightwatchjs.org), [nightwatch-cucumber](https://github.com/mucsi96/nightwatch-cucumber) and [cucumber.js](https://github.com/cucumber/cucumber-js) to run acceptance tests.
```
Comment (tyom) [inline README.md]: Do we write it now in all lower case?

**Comment by tyom** [comment]
Any reason we have inconsistent case in script names (shell vs JS)?
`copy_screenshots.sh` vs `cucumberReport.js`

File: README.md (hunk 1 above)
Comment (feedmypixel) [reply README.md]: From Mills:

screen shot 2017-09-18 at 14 03 52

Comment (teneightfive) [reply README.md]: Lowercase if in middle of sentence I would say. This looks like the start of a sentence so uppercase `D` might make more sense?

**Comment by teneightfive** [comment]
>Any reason we have inconsistent case in script names (shell vs JS)? `copy_screenshots.sh` vs `cucumberReport.js`

Would it be better to be consistent with the rest of the files in the project and use kebab case?

**Comment by feedmypixel** [comment]
> Would it be better to be consistent with the rest of the files in the project and use kebab case?

I've always avoided hyphens for bash scripts, I'm pretty sure there are compatibility issues with them. Snake case works here I feel.

**Comment by feedmypixel** [comment]
> I've always avoided hyphens for bash scripts, I'm pretty sure there are compatibility issues with them. Snake case works here I feel.

https://unix.stackexchange.com/questions/168221/are-there-problems-with-hyphens-in-functions-aliases-and-executables

**Comment by feedmypixel** [comment]
> I've always avoided hyphens for bash scripts, I'm pretty sure there are compatibility issues with them. Snake case works here I feel.

https://unix.stackexchange.com/questions/168221/are-there-problems-with-hyphens-in-functions-aliases-and-executables

**Comment by tyom** [comment]
> https://unix.stackexchange.com/questions/168221/are-there-problems-with-hyphens-in-functions-aliases-and-executables

Does anyone use ash (busybox), csh, tcsh or dash?

**Comment by feedmypixel** [comment]
> Does anyone use ash (busybox), csh, tcsh or dash?

yeah that article doesn't really act as a _backup_ for why not to use hypens in filenames. I have always used underscores for bash scripts would be my answer. @teneightfive can change this if required?

Interestingly if you `ls -la ~` there are underscores and hypens.

**Comment by feedmypixel** [comment]
> @teneightfive can change this if required?

chaps have updated to `kebab-case` as it feels like consistency with the rest of the codebase is the way to go 👍

**Pull request merged**